packages/unmock-core/src/nock.ts

```diff
@@ -166,58 +170,75 @@ export class DynamicServiceSpec {
     } else {
       this.data = JSONSchemify(maybeStatusCode) as Schema;
     }
-    return this.updater({ data: this.data, statusCode: this.statusCode });
+    const { store, service } = this.updater({
+      data: this.data,
+      statusCode: this.statusCode,
+    });
+
+    return buildFluentNock(store, this.baseUrl, this.name, service);
   }
 }
 
-export const nockify = ({
-  backend,
-  baseUrl,
-  name,
-}: {
-  backend: NodeBackend;
-  baseUrl: string;
-  name?: string;
-}) => {
+type FluentDyamicService = {
+  [k in HTTPMethod]: (endpoint: string) => DynamicServiceSpec;
+} & { state: StateType; spy: ServiceSpy };
+
+const buildFluentNock = (
+  store: ServiceStore,
+  baseUrl: string,
+  name?: string,
+  service?: Service,
+): FluentDyamicService => {
   const dynFn = (method: HTTPMethod, endpoint: string) => ({
     statusCode,
     data,
   }: {
     statusCode: number;
     data: Schema;
   }) =>
-    backend.serviceStore.updateOrAdd({
+    store.updateOrAdd({
       baseUrl,
       method,
       endpoint: endpoint.startsWith("/") ? endpoint : `/${endpoint}`,
       statusCode,
       response: data,
       name,
     });
-  return {
-    get(endpoint: string) {
-      return new DynamicServiceSpec(dynFn("get", endpoint), 200);
-    },
-    head(endpoint: string) {
-      return new DynamicServiceSpec(dynFn("head", endpoint), 200);
-    },
-    post(endpoint: string) {
-      return new DynamicServiceSpec(dynFn("post", endpoint), 201);
-    },
-    put(endpoint: string) {
-      return new DynamicServiceSpec(dynFn("put", endpoint), 204);
-    },
-    patch(endpoint: string) {
-      return new DynamicServiceSpec(dynFn("patch", endpoint), 204);
-    },
-    delete(endpoint: string) {
-      return new DynamicServiceSpec(dynFn("delete", endpoint), 200);
-    },
-    options(endpoint: string) {
-      return new DynamicServiceSpec(dynFn("options", endpoint), 200);
-    },
-    trace(endpoint: string) {
-      return new DynamicServiceSpec(dynFn("trace", endpoint), 200);
-    },
-  };
+  const fluent = Object.entries({
+    get: 200,
+    head: 200,
+    post: 201,
+    put: 204,
+    patch: 204,
+    delete: 200,
+    options: 200,
+    trace: 200,
+  }).reduce(
+    (o, [method, code]) => ({
+      ...o,
+      [method]: (endpoint: string) =>
+        new DynamicServiceSpec(
+          dynFn(method as HTTPMethod, endpoint),
+          code,
+          baseUrl,
+          name,
+        ),
+    }),
+    {},
+  ) as FluentDyamicService;
+  if (service !== undefined) {
+    fluent.state = service.state;
+    fluent.spy = service.spy;
```

Comment:
If you add new properties to service, you also have to remember to add them here I assume?